dynawo/sources/Models/CPP/Controls/Shunts/ModelCentralizedShuntsSectionControl/DYNModelCentralizedShuntsSectionControl.h

```diff
@@ -0,0 +1,282 @@
+  /**
+   * @copydoc ModelCPP::initParams()
+   */
+  void initParams() { /* not needed */ }
+  /**
+   * @brief initialize the model from data interface
+   *
+   * @param data data interface to use to initialize the model
+   */
+  void initializeFromData(const boost::shared_ptr<DataInterface> &data);
+  /**
+   * @copydoc ModelCPP::initializeStaticData()
+   */
+  void initializeStaticData() { /* not needed */ }
+  /**
+   * @brief Coherence check on data (asserts, min/max values, sanity checks)
+   */
+  void checkDataCoherence(const double /*t*/) { /* not needed */ }
+  /**
+   * @brief calculate calculated variables
+   */
+  void evalCalculatedVars() { /* not need */ }
+  /**
+   * @brief evaluate the jacobian associated to a calculated variable
+   *
+   * @param iCalculatedVar index of the calculated variable
+   * @param res values of the jacobian
+   */
+  void evalJCalculatedVarI(unsigned iCalculatedVar, std::vector<double> &res) const;
+  /**
+   * @brief  ModelCentralizedShuntsSectionControl transposed jacobian evaluation
+   *
+   * Get the sparse transposed jacobian \f$ Jt=@F/@y' \f$
+   *
+   * @param t Simulation instant
+   * @param cj Jacobian prime coefficient
+   * @param jt jacobian matrix to fullfill
+   * @param rowOffset offset to use to identify the row where data should be
+   * added
+   */
+  void evalJtPrim(const double t, const double cj, SparseMatrix &jt, const int rowOffset);
+  /**
+   * @brief Model mode change type evaluation
+   *
+   * Set the mode change type value depending on current simulation instant and
+   * current state variables values.
+   * @param[in] t Simulation instant
+   * @return mode change type value
+   */
+  modeChangeType_t evalMode(const double t);
+  /**
+   * @brief get the global indexes of the variables used to compute a calculated
+   * variable
+   *
+   * @param iCalculatedVar index of the calculated variable
+   * @param indexes vector to fill with the indexes
+   *
+   */
+  void getIndexesOfVariablesUsedForCalculatedVarI(unsigned iCalculatedVar, std::vector<int> &indexes) const;
+  /**
+   * @copydoc ModelCPP::setGequations()
+   */
+  void setGequations();
+  /**
+   * @brief evaluate the value of a calculated variable
+   *
+   * @param iCalculatedVar index of the calculated variable
+   *
+   * @return value of the calculated variable
+   */
+  double evalCalculatedVarI(unsigned iCalculatedVar) const;
+  /**
+   * @brief ModelCentralizedShuntsSectionControl elements initializer
+   *
+   * Define elements for this model (elements to be seen by other models)
+   *
+   * @param elements  Reference to elements' vector
+   * @param mapElement Map associating each element index in the elements vector
+   * to its name
+   */
+  void defineElements(std::vector<Element> &elements, std::map<std::string, int> &mapElement);
+
+ private:
+  int nbShunts_;                  ///< number of shunts
+  std::vector<int> sections0_;    ///< initial section values of each the shunts
```

Comment:
typo: of each shunt or for each of  the shunts

Reply:
thanks, fixed!